### PR TITLE
Add workflow stubtests for macos

### LINKED ISSUE #5380: Stubtest CI doesn't run on macos platform
Wondering why it currently only runs on linux and windows.

### PR DESCRIPTION
**Pull request by hatal175**
Fixes #5380

I've added several sys.platform tests where I could find some reference that the functions don't exist under macos.
I've also fixed a few simple parameter mismatches on the way.

### CODE REVIEW

File: stdlib/os/__init__.pyi [hunk 1]
```diff
@@ -117,7 +117,9 @@ if sys.platform != "win32":
     RTLD_LOCAL: int
     RTLD_NODELETE: int
     RTLD_NOLOAD: int
-    RTLD_DEEPBIND: int
+
+    if sys.platform != "darwin":
+        RTLD_DEEPBIND: int
```
Comment (Akuli) [inline stdlib/os/__init__.pyi:122]: This is linux-specific (actually gnu libc specific, but typeshed doesn't really support GNU Hurd yet, lol). `dlopen(3)` linux man page says: "The RTLD_NOLOAD, RTLD_NODELETE, and RTLD_DEEPBIND flags are GNU  extensions; the first two of these flags are also present on Solaris."

File: stdlib/posix.pyi [hunk 2]
```diff
@@ -59,8 +60,9 @@ F_TEST: int
 F_TLOCK: int
 F_ULOCK: int
 
-GRND_NONBLOCK: int
-GRND_RANDOM: int
+if sys.platform != "darwin":
+    GRND_NONBLOCK: int
+    GRND_RANDOM: int
```
Comment (Akuli) [inline stdlib/posix.pyi:65]: These are linux-specific: they are for `getrandom(2)`, and that manual page says it's linux-specific.

File: stdlib/posix.pyi [hunk 3]
```diff
@@ -99,21 +102,24 @@ P_ALL: int
 P_PGID: int
 P_PID: int
 
-RTLD_DEEPBIND: int
+if sys.platform != "darwin":
+    RTLD_DEEPBIND: int
```
Comment (Akuli) [inline stdlib/posix.pyi:106]: Linux-specific again. (Somehow this is specified in two different modules)

File: stdlib/posix.pyi [hunk 4]
```diff
@@ -99,21 +102,24 @@ P_ALL: int
...
 SCHED_FIFO: int
-SCHED_IDLE: int
 SCHED_OTHER: int
-SCHED_RESET_ON_FORK: int
 SCHED_RR: int
 
+if sys.platform != "darwin":
+    SCHED_BATCH: int
+    SCHED_IDLE: int
+    SCHED_RESET_ON_FORK: int
```
Comment (Akuli) [inline stdlib/posix.pyi:121]: `sched_setscheduler(2)` says "The SCHED_BATCH and SCHED_IDLE policies are Linux-specific."

File: stdlib/selectors.pyi [hunk 5]
```diff
@@ -41,12 +41,13 @@ if sys.platform != "win32":
...
         def select(self, timeout: Optional[float] = ...) -> List[Tuple[SelectorKey, _EventMask]]: ...
         def get_map(self) -> Mapping[FileDescriptorLike, SelectorKey]: ...
-    class EpollSelector(BaseSelector):
-        def fileno(self) -> int: ...
-        def register(self, fileobj: FileDescriptorLike, events: _EventMask, data: Any = ...) -> SelectorKey: ...
-        def unregister(self, fileobj: FileDescriptorLike) -> SelectorKey: ...
-        def select(self, timeout: Optional[float] = ...) -> List[Tuple[SelectorKey, _EventMask]]: ...
-        def get_map(self) -> Mapping[FileDescriptorLike, SelectorKey]: ...
+    if sys.platform != "darwin":
+        class EpollSelector(BaseSelector):
```
Comment (Akuli) [inline stdlib/selectors.pyi:45]: `epoll(7)` says "The  epoll  API  is Linux-specific."

File: stdlib/time.pyi [hunk 6]
```diff
@@ -14,7 +14,8 @@ timezone: int
 tzname: Tuple[str, str]
 
 if sys.version_info >= (3, 7) and sys.platform != "win32":
-    CLOCK_BOOTTIME: int  # Linux
+    if sys.platform != "darwin":
+        CLOCK_BOOTTIME: int  # Linux
```
Comment (Akuli) [inline stdlib/time.pyi:18]: Should check for linux

File: stdlib/os/__init__.pyi [hunk 7]
```diff
@@ -791,8 +793,9 @@ else:
     def spawnvp(mode: int, file: AnyPath, args: _ExecVArgs) -> int: ...
     def spawnvpe(mode: int, file: AnyPath, args: _ExecVArgs, env: _ExecEnv) -> int: ...
     def wait() -> Tuple[int, int]: ...  # Unix only
-    from posix import waitid_result
-    def waitid(idtype: int, ident: int, options: int) -> waitid_result: ...
+    if sys.platform != "darwin":
+        from posix import waitid_result
+        def waitid(idtype: int, ident: int, options: int) -> waitid_result: ...
```
Comment (Akuli) [inline stdlib/os/__init__.pyi:798]: Can we just re-export `posix.waitid`?

```suggestion
    if sys.platform != "darwin":
        from posix import waitid as waitid
```

Comment (hatal175) [reply stdlib/os/__init__.pyi:798]: Waitid is not defined in posix stub

**Review by Akuli — APPROVED** [review]

**Pull request merged**